Core/Core/Database/CoreData/Models/ModuleItem.swift

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright (C) 2019-present Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CoreData
+
+private let encoder = JSONEncoder()
+private let decoder = JSONDecoder()
+
+public class ModuleItem: NSManagedObject {
+    @NSManaged public var id: String
+    @NSManaged public var moduleID: String
+    @NSManaged public var position: Int
+    @NSManaged public var title: String
+    @NSManaged public var indent: Int
+    @NSManaged public var htmlURL: URL
+    @NSManaged public var url: URL?
+    @NSManaged public var published: Bool
+    @NSManaged public var module: Module?
+    @NSManaged public var primitiveType: Data?
+    public var type: ModuleItemType? {
+        get {
+            willAccessValue(forKey: "type")
```

Comment:
why do we need the explicit will & did?

Reply:
They are needed to trigger change notifications and [setting primitive values will not call them](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CoreData/LifeofaManagedObject.html).

I was thinking `primitiveType` is want I want here but we've been using `fooRaw` everywhere else. I guess I could change this to `typeRaw` and have a custom getter and setter through `type` and not bother with `willAccessValue` and `didAccessValue`.

Reply:
I cleaned this up with `typeRaw` instead of `primitiveType`.